### ISSUE TITLE
Nowy zasób - moje oferty w wersji beta

### ISSUE THREAD

**Issue by MartaNowaczyk** [allegro/allegro-api#794]
Udostępniliśmy **w wersji beta zasób do pobierania listy ofert** zautoryzowanego sprzedawcy, który utworzył te oferty:

- [GET /sale/offers](https://developer.allegro.pl/documentation/#/offer-management/searchOffersUsingGET) - chcę pobrać listę ofert ze szczegółami.

Jednym zapytaniem otrzymasz listę ofert z najważniejszymi polami, które wprowadziłeś do oferty. Jest to odpowiednik zakładki [Moje oferty](https://allegro.pl/seller-offers).

Stworzyliśmy też [poradnik - jak zarządzać ofertami](https://developer.allegro.pl/my_offers/) przez REST API. 

Zapoznajcie się z nim i czekamy na wasze komentarze i uwagi.

**Comment by rtnet-pl** [comment]
Mam pytanie - saleInfo.biddersCount - jaki obejmuje okres? Czy są to wszystkie zakupy od początku istnienia oferty?
Brakuje mi pola wskazującego czy oferta ma SMART.

**Comment by MartaNowaczyk** [comment]
rtnet-pl pole o którym piszesz dotyczy licytacji i jest toliczba osób, która aktualnie licytuje, czyli osób, które złożyły ofertę kupna w danej ofercie licytacji.

Pole z oznaczeniem SMART - nie ma takiego oznaczenia na froncie, w REST API też nie mamy na ten moment w planach, ale dzięki za sugestię.

**Comment by andrzejg07** [comment]
pogratulować metody jest ok - brak tylko identyfikatora cennika dostaw czy jest przewidywany ?

**Comment by mapsi-fixi** [comment]
Metoda nie podaje poprawnej ilości przedmiotów sprzedanych, jest to zawsze 0

**Comment by MartaNowaczyk** [comment]
@andrzejg07 - tak mamy w planach
@mapsi-fixi - możesz podać jakiś przykład oferty? Sprawdziłam przed chwilą i zwraca mi poprawny stock.sold

**Comment by slawomirponiewaz** [comment]
Czy są jakieś problemy z tym zasobem?
Obecnie otrzymuję: "Cannot read property '1' of null"

**Comment by slawomirponiewaz** [comment]
Czy obecnie obecnie występują problemy z tym zasobem ?
Otrzymuję zwrotnie taka wiadomość: "Cannot read property '1' of null"

**Comment by slawomirponiewaz** [comment]
Czy jest jakiś problem z tym zasobem ?
Obecnie otrzymuję taki komunikat: "Cannot read property '1' of null"

**Comment by MartaNowaczyk** [comment]
Sprawdzałam i działa. Czy problem u ciebie nadal występuje?

**Comment by SoftSolid** [comment]
Czy parametr "limit" działa ? Próbuję pobrać więcej niż standardowe 20 sztuk i nie ma żadnego efektu.

**Comment by mapsi-fixi** [comment]
@MartaNowaczyk tak między innymi oferta o id 6205660179, btw wszystkie oferty na tym koncie pod wartością sold mają 0, oczywiście mówimy o sandboxie

**Comment by mapsi-fixi** [comment]
@SoftSolid jeżeli chodzi o sandbox to u mnie dziła poprawnie

**Comment by SoftSolid** [comment]
Na produkcyjnym u mnie nie działa, chyba że źle podaję limit parametr limit.

**Comment by MartaNowaczyk** [comment]
@SoftSolid Na produkcji bez problemu otrzymuję więcej niż 20 ofert. Przykładowe wywołanie z parametrem limit https://api.allegro.pl/sale/offers?limit=100

**Comment by MartaNowaczyk** [comment]
@mapsi-fixi wcześniej sprawdzałam na produkcji i działa. Na sandboxie tak jak ty mam stock.sold=0. Sprawdzamy co się stało. Dam znać.

**Comment by mapsi-fixi** [comment]
@MartaNowaczyk  Bardzo proszę jeszcze o sprawdzenie dlaczego drafty się nie listują, wczoraj działało.

**Comment by MartaNowaczyk** [comment]
@mapsi-fixi rozumiem, że też na sandboxie? Dla pewności pisz zawsze gdzie.

**Comment by mapsi-fixi** [comment]
Tak, chodzi o sandbox :)

**Comment by damianSzechlicki** [comment]
Metoda bardzo przydatna. Brakuje mi tylko informacji o dostawie, dokładnie terminu dostawy. Będzie może dodana ta informacja w przyszłości? ;)

**Comment by MartaNowaczyk** [comment]
@mapsi-fixi sprawdzamy te drafty na sandboxie. Czy jak zrobisz draft na froncie, to masz go na liście po wywołaniu tego zasobu? Rozumiem, że dodajesz do URLa requestu wszystkie statusy /sale/offers?publication.status=INACTIVE,ACTIVATING,ACTIVE,ENDED

**Comment by MartaNowaczyk** [comment]
@damianSzechlicki mamy w planach dodanie id cennika dostaw

**Comment by mapsi-fixi** [comment]
jeżeli utworze draft na froncie to nie widać go poprzez api jak i również na froncie sandboxa. To sam się tyczy draftów utworzonych przez api. Natomiast gdy utworzę taki draft to mogę go edytować jeżeli podam jego id w metodzie GET sale/offers/{id}.
Zapytanie wysyłam takie jak podałaś wyżej czyli  /sale/offers?publication.status=INACTIVE,ACTIVATING,ACTIVE,ENDED

**Comment by MartaNowaczyk** [comment]
@mapsi-fixi dzięki za informację. U mnie draft utworzony na froncie jest widoczny, gorzej z tym utworzonym przez REST API. Sprawdzamy.

**Comment by mapsi-fixi** [comment]
@MartaNowaczyk u mnie na początku też sie dodawały i były widoczne, być może chodzi o ich ilość?

**Comment by MartaNowaczyk** [comment]
@mapsi-fixi Dzięki za info. Sprawdzamy różne scenariusze.

**Comment by SoftSolid** [comment]
7367088933 - jest 15 sztuk, zasób zwraca 17. Dlaczego ? Stan z godziny 23.49

**Comment by MartaNowaczyk** [comment]
@SoftSolid postaram się ustalić czy coś jest na rzeczy, gdybyś miał kolejny taki przykład podaj proszę
@dostawcaopon sprawdzimy limity

**Comment by mapsi-fixi** [comment]
@MartaNowaczyk czy problemy zgłoszone przeze mnie są już znane ?

**Comment by MartaNowaczyk** [comment]
@SoftSolid dzięki za info
@mapsi-fixi  jeszcze nie znamy przyczyny, ale pracujemy nad tym

**Comment by MartaNowaczyk** [comment]
@dostawcaopon przepraszam, że ci nie odpisałam. Wczoraj poprawiliśmy konfigurację, także nie powinieneś mieć problemu z limitem. Jeśli jest inaczej, daj znać.

**Comment by SoftSolid** [comment]
7604625377 - nie mogę wznowić tej aukcji. Ustawiłem stan magazynowy na 1 ale aukcja się nie wznawia.

**Comment by MartaNowaczyk** [comment]
@SoftSolid jaki dostajesz błąd? Możesz podesłać curle?

**Comment by SoftSolid** [comment]
Problem w tym, że nic nie dostaję. Pusty. Na pewno chodzi o parametry bo zmiana parametrów z poziomu allegro umożliwiła wystawienie aukcji.

**Comment by MartaNowaczyk** [comment]
@SoftSolid w takim razie powinieneś dostać błąd - fajnie by było jakbyś trace-id miał. W odpowiedzi nagłówków nie dostajesz? Jeśli trafisz na kolejny taki przypadek podeślij proszę chociaż zapytanie, bo dobrze by było sprawdzić ten przypadek.

**Comment by mjag8** [comment]
Metoda nie podaje poprawnej ilości przedmiotów sprzedanych, moja przykładowa oferta ma dostępnych 7 sztuk, łącznie liczba sztuk w edycji oferty to 18. Sprzedanych było 11.

Metoda zwraca:
      ["stock"]=>
      object(stdClass)#267 (2) {
        ["available"]=>
        int(7)
        ["sold"]=>
        int(7)
      }

**Comment by MartaNowaczyk** [comment]
@mjag8 Podaj proszę numer tej oferty?

**Comment by mjag8** [comment]
Oferta nr 7460121046

Teraz rozumiem, chyba o co chodzi. Ale wciąż nie widzę możliwości określenia, jaka ilość została sprzedana przez cały okres trwania oferty.

**Comment by MartaNowaczyk** [comment]
@mjag8 w REST API zwracamy liczbę przedmiotów z ostatnich 30 dni

**Comment by mjag8** [comment]
Tak, rozumiem już. A czy planujecie wprowadzić gdzieś pole zwracające ilość sprzedanych podczas całego okresu trwania oferty? Tak na prawdę nie ma możliwości określenia ile sztuk w sumie się wystawiło nawet.
W edycji ręcznej oferty jest to
Formaty sprzedaży -> input Liczba sztuk

**Comment by mjag8** [comment]
Aha
No cóż, w samym API to może nie aż tak bardzo. Ale na potrzeby działania całego danego systemu byłoby to dla mnie krokiem w przód. W tym momencie nie jestem w stanie obsługiwać w całości spraw typu rozdzielenie odpowiedniej ilości nakładu dla sprzedaży allegro, tworzenie statystyk i raportów. No bo nie ma możliwości w 100% stwierdzić jaki nakład łącznie allegro przyjęło i znalazł się w ofercie.
Ale to moja sugestia

**Comment by MartaNowaczyk** [comment]
Dzięki za argumenty. Przekażę twoją sugestię do odpowiedniego zespołu. Na ten moment pozostaje zbieranie takiej informacji po swojej stronie.

**Comment by lukasnet** [comment]
Brak możliwości sortowania po tytule

**Comment by MartaNowaczyk** [comment]
@lukasnet tak, jest tylko filtrowanie po tytule. Czy ta opcja jest ci niezbędna?

**Comment by lukasnet** [comment]
Tak, przydałaby się, korzystając obecnie z webapi jest to dla mnie domyślna opcja sortowania i łatwiejsze zarządzanie listą ofert.

**Comment by MartaNowaczyk** [comment]
@lukasnet sugestię zgłosiłam do odpowiedniego zespołu, ale nic nie obiecuję, bo nie przypominam sobie, żeby jeszcze ktoś zgłaszał prośbę o sortowanie po tytule. Jeśli dodamy taką możliwość damy informację w oddzielnym komunikacie.
